lib/animation.js

```diff
@@ -78,492 +50,499 @@ Animation.TemporalFallback.prototype.stop = function() {
-          return accum;
-        }, {});
+      // Pad the right side of keyFrames arrays with null
+      for (let i = keyFrames.length; i < cuePoints.length; i++) {
+        keyFrames.push(null);
       }
-    }
 
-    return calcValue;
-  }, this);
+      keyFrames.forEach((keyFrame, i, source) => {
 
-  return result;
-};
+        if (keyFrame !== null) {
 
-// Make sure our keyframes conform to a standard
-Animation.prototype.normalizeKeyframes = function() {
+          // keyFrames need to be converted to objects
+          if (typeof keyFrame !== "object") {
+            keyFrame = {
+              step: keyFrame,
+              easing: "linear"
+            };
+          }
 
-  var previousVal,
-    keyFrameSet = Fn.cloneDeep(this.keyFrames),
-    cuePoints = this.cuePoints;
+          // Replace step values
+          if (typeof keyFrame.step !== "undefined") {
+            keyFrame.value = keyFrame.step === false ?
+              previousVal : previousVal + keyFrame.step;
+          }
 
-  // Run through the target's normalization
-  keyFrameSet = this.target[Animation.normalize](keyFrameSet);
+          // Set a default easing function
+          if (!keyFrame.easing) {
+            keyFrame.easing = "linear";
+          }
 
-  // keyFrames can be passed as a single dimensional array if
-  // there is just one servo/device. If the first element is not an
-  // array, nest keyFrameSet so we only have to deal with one format
-  if (!Array.isArray(keyFrameSet[0])) {
-    keyFrameSet = [keyFrameSet];
-  }
+          // Copy value from another frame
+          /* istanbul ignore if */
+          if (typeof keyFrame.copyValue !== "undefined") {
+            keyFrame.value = source[keyFrame.copyValue].value;
+          }
 
-  keyFrameSet.forEach(function(keyFrames) {
+          // Copy everything from another keyframe in this array
+          /* istanbul ignore if */
+          if (keyFrame.copyFrame) {
+            keyFrame = source[keyFrame.copyFrame];
+          }
 
-    // Pad the right side of keyFrames arrays with null
-    for (var i = keyFrames.length; i < cuePoints.length; i++) {
-      keyFrames.push(null);
-    }
+          previousVal = keyFrame.value;
 
-    keyFrames.forEach(function(keyFrame, i, source) {
+        } else {
 
-      if (keyFrame !== null) {
+          if (i === source.length - 1) {
+            keyFrame = {
+              value: previousVal,
+              easing: "linear"
+            };
+          } else {
+            keyFrame = null;
+          }
 
-        // keyFrames need to be converted to objects
-        if (typeof keyFrame !== "object") {
-          keyFrame = {
-            step: keyFrame,
-            easing: "linear"
-          };
         }
+        source[i] = keyFrame;
 
-        // Replace step values
-        if (typeof keyFrame.step !== "undefined") {
-          keyFrame.value = keyFrame.step === false ?
-            previousVal : previousVal + keyFrame.step;
-        }
+      }, this);
```

Comment:
I think the "this" binding is unnecessary since we're using an arrow function on the opening forEach.